File: scripts/db-push.mjs
import { spawn } from 'child_process';
import { createInterface } from 'readline';

function runCommand(cmd, args, autoRespond = false) {
  return new Promise((resolve, reject) => {
    const proc = spawn(cmd, args, {
      cwd: '/home/ubuntu/kinga-replit',
      stdio: ['pipe', 'pipe', 'pipe'],
      env: { ...process.env, FORCE_COLOR: '0' }
    });
    
    let output = '';
    
    proc.stdout.on('data', (data) => {
      const text = data.toString();
      output += text;
      process.stdout.write(text);
      
      // Auto-answer "create column" / "create table" prompts
      if (autoRespond && (text.includes('created or renamed') || text.includes('create column') || text.includes('create table'))) {
        setTimeout(() => {
          proc.stdin.write('\n');
        }, 100);
      }
    });
    
    proc.stderr.on('data', (data) => {
      const text = data.toString();
      output += text;
      process.stderr.write(text);
      
      if (autoRespond && (text.includes('created or renamed') || text.includes('create column') || text.includes('create table'))) {
        setTimeout(() => {
          proc.stdin.write('\n');
        }, 100);
      }
    });
    
    proc.on('close', (code) => {
      resolve({ code, output });
    });
    
    proc.on('error', reject);
    
    // Safety timeout
    setTimeout(() => {
      proc.kill();
      resolve({ code: -1, output });
    }, 120000);
  });
}

console.log('=== Running drizzle-kit generate ===');
const genResult = await runCommand('npx', ['drizzle-kit', 'generate'], true);
console.log(`\n=== Generate exit code: ${genResult.code} ===\n`);

console.log('=== Running drizzle-kit migrate ===');
const migResult = await runCommand('npx', ['drizzle-kit', 'migrate'], false);
console.log(`\n=== Migrate exit code: ${migResult.code} ===`);

File: scripts/db-push.py
#!/usr/bin/env python3
"""
Auto-answer drizzle-kit generate prompts by selecting 'create column' (first option)
for all rename/create questions, then run migrate.
"""
import pexpect
import sys
import os

os.chdir("/home/ubuntu/kinga-replit")

# Run drizzle-kit generate
print("=== Running drizzle-kit generate ===")
child = pexpect.spawn("npx drizzle-kit generate", timeout=120, encoding="utf-8")
child.logfile_read = sys.stdout

while True:
    try:
        # Wait for either a prompt or EOF
        index = child.expect([
            r"created or renamed from another column\?",  # Column rename question
            r"created or renamed from another table\?",   # Table rename question
            pexpect.EOF,
            pexpect.TIMEOUT
        ], timeout=60)
        
        if index == 0 or index == 1:
            # Press Enter to select first option (create)
            child.sendline("")
        elif index == 2:
            # Process finished
            break
        elif index == 3:
            print("\nTIMEOUT waiting for prompt")
            break
    except Exception as e:
        print(f"\nException: {e}")
        break

child.close()
print(f"\n=== Generate exit status: {child.exitstatus} ===")

# Run drizzle-kit migrate
print("\n=== Running drizzle-kit migrate ===")
child2 = pexpect.spawn("npx drizzle-kit migrate", timeout=120, encoding="utf-8")
child2.logfile_read = sys.stdout

child2.expect(pexpect.EOF, timeout=120)
child2.close()
print(f"\n=== Migrate exit status: {child2.exitstatus} ===")
